config/rootrc.in

```diff
@@ -514,103 +514,13 @@ Url.Special:         file: hpss: dcache:
-# XNet.ConnectDomainDenyRE
-#                       - sequence of TRegexp regular expressions
-#                         separated by a |.
-#                         A domain is denied access to for the
-#                         first connection if it matches one of these
-#                         regexps. Example:
-#                         slac.stanford.edu|pd.infn.it|fe.infn.it
-# XNet.RedirDomainAllowRE
-#                       - sequence of TRegexp regular expressions
-#                         separated by a |.
-#                         A domain is granted access to for a
-#                         redirection if it matches one of these
-#                         regexps. Example:
-#                         slac.stanford.edu|pd.infn.it|fe.infn.it
-# XNet.RedirDomainDenyRE
-#                       - sequence of TRegexp regular expressions
-#                         separated by a |.
-#                         A domain is denied access to for a
-#                         redirection if it matches one of these
-#                         regexps. Example:
-#                         slac.stanford.edu|pd.infn.it|fe.infn.it
-# XNet.MaxRedirectCount - maximum number of redirections from
-#                         server [16]
-# XNet.ReconnectWait    - sleep-time before going back to the
-#                         load balancer (or rebouncing to the same
-#                         failing host) after a read/write error
-#                         [5 s]
-# XNet.ReadAheadSize    - read ahead value, in bytes, to use with the XrdClient
-#                         internal caching scheme
-#                         Generally not much useful together with the TFile cache,
-#                         which will disable the readahead as soon it starts operating,
-#                         i.e. requesting async data through XrdClient.
-#                         Note that we cannot suppose that the TFile cache
-#                         will start working at the beginning of the data processing.
-#                         Setting a read ahead size > 0 will involve this phase,
-#                         sometimes it could be an advantage, sometimes not.
-#                         If > 0, TXNetFile will switch it on again if it sees that
-#                         synchronous ReadBuffer requests are issued again
-
-XNet.ReadAheadSize: 0
-
-# XNet.ReadCacheSize    - Default read cache size inside XrdClient, in bytes.
-#                         TXNetFile/TFileReadCache automatically sets this to a
-#                         'correct' value when needed.
-#                         Set a value here to override a *minimum*
-#                         size. It is not preallocated.
-#                         TFile/TXNetFile can anyway make it grow
-#                         automatically when needed.
-
-XNet.ReadCacheSize: 10000000
-
-# XNet.FirstConnectMaxCnt
-#                       - Number of connect retries to the first host
```

Comment:
Just a comment: this is also referred to in `proof`:
https://github.com/root-project/root/blob/master/proof/proofd/src/XrdProofdProofServMgr.cxx#L3816

In the future, we also have to check how much of proof is also dead code now.

Reply:
If I analyzed the surrounding code correctly, this is only for *setting* the variable but it is never read (which admittedly makes it dead code as well...). The only variables actually read from `proof` are `XNet.Debug` and `XNet.PrintTAG` as well as `XNet.SOCKS4Host` and `XNet.SOCKS4Port` (the latter two being internal variables that are set by `TProof`).

Reply:
I see. It's definitely dead code, because the `xproofd` option was removed:
https://github.com/root-project/root/blob/master/cmake/modules/RootBuildOptions.cmake#L400

And the inclusion of the `proofd` directory is conditional on that option:
https://github.com/root-project/root/blob/master/proof/CMakeLists.txt#L12

The `proof` option was removed in 6.32 as well. That means its maybe time to remove `proof/proofd` and `proof/proofx` as well? If I recall correctly, `proofd` was only running with the old `netx` and old xrootd versions anyway.

Reply:
Yes, looks like we want to remove that dead code as well. And while adding to that list, `net/monalisa` should also go I think...